dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -1360,10 +1371,21 @@ private String getFromSubQueryLimitAndOffset( TrackedEntityInstanceQueryParams p
     {
         StringBuilder limitOffset = new StringBuilder();
         int limit = params.getMaxTeiLimit();
+        int teiQueryLimit = Integer
+            .parseInt( instanceStoreServicesSupplier.getConfigurationProvider()
```

Comment:
It seems to me that we do not seem to validate the config and fail DHIS2 on startup if the config is invalid. This is unfortunate since a user can put in a String like `foo` into this config and it will only manifest when the first user makes a request that calls this store. If we do have the ability to validate the config and fail DHIS2 startup then lets do it 😄 If not, can we at least parse this only once maybe in the constructor and store it in a field?